### PR TITLE
ethtool: add several link modes

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/shared/ethtool-util.c [hunk 1]
```diff
@@ -112,21 +112,21 @@ static const char* const ethtool_link_mode_bit_table[] = {
...
+        [ETHTOOL_LINK_MODE_100000baseKR2_Full_BIT]         = "100000basekr2-full",
+        [ETHTOOL_LINK_MODE_100000baseSR2_Full_BIT]         = "100000basesr2-full",
+        [ETHTOOL_LINK_MODE_100000baseCR2_Full_BIT]         = "100000basecr2-full",
+        [ETHTOOL_LINK_MODE_100000baseLR2_ER2_FR2_Full_BIT] = "100000baselr2-er2-fr2-full",
+        [ETHTOOL_LINK_MODE_100000baseDR2_Full_BIT]         = "100000basedr2-full",
+        [ETHTOOL_LINK_MODE_200000baseKR4_Full_BIT]         = "200000basekr4-full",
+        [ETHTOOL_LINK_MODE_200000baseSR4_Full_BIT]         = "200000basesr4-full",
+        [ETHTOOL_LINK_MODE_200000baseLR4_ER4_FR4_Full_BIT] = "200000baselr4-er4-fr4-full",
+        [ETHTOOL_LINK_MODE_200000baseDR4_Full_BIT]         = "200000basedr4-full",
+        [ETHTOOL_LINK_MODE_200000baseCR4_Full_BIT]         = "200000basecr4-full",
```
Comment (poettering) [inline src/shared/ethtool-util.c:129]: hmm, the underscored names have already been relesed with v243, no? changing them is a compat breakage no?

at the very least probably needs some NEWS entry.

other option might be to add a tweak to the parser for this enum, and first convert all underscore to dash?

Comment (poettering) [reply src/shared/ethtool-util.c:129]: or maybe it's not a problem since we never documented these?

Comment (keszybz) [reply src/shared/ethtool-util.c:129]: Or do what we do with /proc/cmdline and treat "-" and "_" as equivalent?

Comment (yuwata) [reply src/shared/ethtool-util.c:129]: @poettering and @keszybz Sorry for late reply. The renamed values are not documented yet. So, I think it is safe to convert them silently. Moreover, I am skeptical that all values should be documented, as I have no idea in what situations we need to set `Advertise=` in .link file... So, in this PR, I do not update the man page.

Comment (keszybz) [reply src/shared/ethtool-util.c:129]: Yeah, I guess that's reasonable. If anyone complains, we can always add a fixup later ;)

**Pull request merged**